openpype/hosts/houdini/api/lib.py

```diff
@@ -1054,3 +1054,72 @@ def add_self_publish_button(node):
     template = node.parmTemplateGroup()
     template.insertBefore((0,), button_parm)
     node.setParmTemplateGroup(template)
+
+
+def getSceneViewer():
+    """
+    return an instance of a visible viewport.
+    There may be many, some could be closed, any visible are current
+    """
+    panes = hou.ui.paneTabs()
+    panes = [x for x in panes if x.type() == hou.paneTabType.SceneViewer]
+    panes = sorted(panes, key=lambda x: x.isCurrentTab())
+    if panes:
+        return panes[-1]
+    else:
+        return None
+
+
+def sceneview_snapshoot(
+        filename="$HIP/thumbnails/$HIPNAME.$F4.jpg",
+        fstart=None,
+        fend=None):
+    """
+    It takes snapshoot of your scene view fot the given frame range.
+    So, it's capable of generating snapshoots image sequence.
+    It works in different Houdini context e.g. Objects, Solaris
+
+    Example:
+        from openpype.hosts.houdini.api import lib
+        lib.sceneview_snapshoot()
+
+    Notes:
+        .png output will render poorly, so use .jpg.
+
+        How it works:
+            Get the current sceneviewer (may be more than one or hidden)
+            and screengrab the perspective viewport to a file in the
+            publish location to be picked up with the publish.
+
+        Credits:
+            https://www.sidefx.com/forum/topic/42808/?page=1#post-354796
+
+    Args:
+        filename (str): filename with the desired full path.
+        fstart (int): the frame at which snapshooting starts
+        fend (int): the frame at which snapshooting ends
+    """
+
+    try:
```

Comment:
Why it is in try except? There is no way how source caller can know it crashed.

Reply:
This function feels like somewhat of a matching logic to this [here](https://github.com/ynput/OpenPype/pull/5425/files#diff-82c5362a4187d4f702b172c3d1c22f711aa718eb624d45aec98665ec72909d9dR296-R333).

Which might be a good reference as well.


And I completely agree, remove the `try..except` here.

Reply:
yes, It's the same logic 😅.. 
I'm going to check that PR maybe I can steal some code from it.